### PR TITLE
AE-453: Monitor mint() calls on VotingToken

### PR DESCRIPTION
**Pull request by echoAE**
This handler checks the trace data for transactions to determine if any address other than the authorized Voting contract address calls the mint() method on the UMA VotingToken contract.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,12 @@ the following handlers:
   - Type is always set to "suspicious"
   - Metadata field contains to and from addresses
 
+- AE-UMA-UNAUTHORIZED-MINT
+  - Fired when the UMA VotingToken.mint() method is called by any address other than the UMA Voting contract
+  - Severity is always critical
+  - Type is always set to exploit
+  - Metadata field contains VotingToken contract address, caller address, and transaction hash
+
```
Comment (arbitrary-cr0) [inline README.md:34]: We may want to also mention that this agent needs traces enabled

**Review by arbitrary-cr0 — CHANGES_REQUESTED** [review]
Minor updates to `README.md`

File: src/agent.js [hunk 2]
```diff
@@ -7,7 +8,13 @@ const handleTransaction = async (txEvent) => {
...
   ]);
 
+  const [monitorMintCallsFindings] = await Promise.all([
+    monitorMintCallsAgent.handleTransaction(txEvent),
+  ]);
+
   findings.push(...deployerWatchFindings);
+  findings.push(...monitorMintCallsFindings);
+
   return findings;
```
Comment (y0wl) [inline src/agent.js:18]: this should not be in a separate `Promise.all`. it should be in the same guy as `deployerWatchAgent`

I suggest you remove the array capturing thing with pushing the spread arrays and just use return the flat result

```
> a = [[1],[2],[3]]
[ [ 1 ], [ 2 ], [ 3 ] ]
> a.flat()
[ 1, 2, 3 ]
```

```
  const findings = (
    await Promise.all([
      deployerWatchAgent.handleTransaction(txEvent),
      monitorMintCallsAgent.handleTransaction(txEvent),
    ]);
  ).flat()```

or something along the lines of that...

File: src/monitor-mint-calls.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+} = require('forta-agent');
+
+// load required shared types
+const {
+  VotingToken: votingTokenAddress,
+  Voting: votingAddress,
+} = require('../contract-addresses.json');
+
+const { abi: votingTokenAbi } = require('../abi/VotingToken.json');
+const { umaEverestId } = require('../agent-config.json');
```
Comment (y0wl) [inline src/monitor-mint-calls.js:14]: new NPM abi method might wanna switch it up before merge

File: src/monitor-mint-calls.spec.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+const { handleTransaction, createAlert } = require('./monitor-mint-calls');
+
+// load required contract addresses
+const {
+  VotingToken: votingTokenAddress,
+  Voting: votingAddress,
+} = require('../contract-addresses.json');
+
+// get the abi for the VotingToken contract
+const { abi: votingTokenAbi } = require('../abi/VotingToken.json');
```
Comment (y0wl) [inline src/monitor-mint-calls.spec.js:18]: NPM method maybe

**Review by y0wl — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -26,6 +26,12 @@ the following handlers:
   - Type is always set to "suspicious"
   - Metadata field contains to and from addresses
 
+- AE-UMA-UNAUTHORIZED-MINT
+  - Fired when the UMA VotingToken.mint() method is called by any address other than the UMA Voting contract
+  - Severity is always critical
```
[the remark this answers is not in the document]
Comment (echoAE) [reply README.md]: Double quotes added.

File: README.md (hunk 1 above)
Comment (echoAE) [reply README.md:35]: Good idea, added.

File: src/agent.js (hunk 2 above)
Comment (echoAE) [reply src/agent.js]: Good catch.  I consolidated the promises into one Promise.all() the way you suggested (it appeared to function appropriately in `npm test` and `npm start`).

File: src/monitor-mint-calls.js (hunk 3 above)
Comment (echoAE) [reply src/monitor-mint-calls.js]: Added import of `@uma/contracts-node` module and associated function calls.

File: src/monitor-mint-calls.spec.js (hunk 4 above)
Comment (echoAE) [reply src/monitor-mint-calls.spec.js]: Added import of `@uma/contracts-node` module and associated function calls.

**Review by y0wl — COMMENTED** [review]
---redacted---

**Review by y0wl — APPROVED** [review]

**Review by arbitrary-cr0 — APPROVED** [review]

**Pull request merged**